nibabel/streamlines/tests/test_array_sequence.py

```diff
@@ -0,0 +1,255 @@
+
+    def test_arraysequence_extend(self):
+        new_data = generate_data(nb_arrays=10,
+                                 common_shape=SEQ_DATA['seq'].common_shape,
+                                 rng=SEQ_DATA['rng'])
+
+        # Extend with an empty list.
+        seq = SEQ_DATA['seq'].copy()  # Copy because of in-place modification.
+        seq.extend([])
+        check_arr_seq(seq, SEQ_DATA['data'])
+
+        # Extend with a list of ndarrays.
+        seq = SEQ_DATA['seq'].copy()  # Copy because of in-place modification.
+        seq.extend(new_data)
+        check_arr_seq(seq, SEQ_DATA['data'] + new_data)
+
+        # Extend with another `ArraySequence` object.
+        seq = SEQ_DATA['seq'].copy()  # Copy because of in-place modification.
+        seq.extend(ArraySequence(new_data))
+        check_arr_seq(seq, SEQ_DATA['data'] + new_data)
+
+        # Extend with an `ArraySequence` view (e.g. been sliced).
+        # Need to make sure we extend only the data we need.
+        seq = SEQ_DATA['seq'].copy()  # Copy because of in-place modification.
+        seq.extend(ArraySequence(new_data)[::2])
+        check_arr_seq(seq, SEQ_DATA['data'] + new_data[::2])
+
+        # Test extending an empty ArraySequence
+        seq = ArraySequence()
+        seq.extend(ArraySequence())
+        check_empty_arr_seq(seq)
+
+        seq.extend(SEQ_DATA['seq'])
+        check_arr_seq(seq, SEQ_DATA['data'])
+
+        # Extend with elements of different shape.
+        data = generate_data(nb_arrays=10,
+                             common_shape=SEQ_DATA['seq'].common_shape*2,
+                             rng=SEQ_DATA['rng'])
+        seq = SEQ_DATA['seq'].copy()  # Copy because of in-place modification.
+        assert_raises(ValueError, seq.extend, data)
+
+    def test_arraysequence_getitem(self):
+        # Get one item
+        for i, e in enumerate(SEQ_DATA['seq']):
+            assert_array_equal(SEQ_DATA['seq'][i], e)
+
+        # Get all items using indexing (creates a view).
+        indices = list(range(len(SEQ_DATA['seq'])))
+        seq_view = SEQ_DATA['seq'][indices]
+        check_arr_seq_view(seq_view, SEQ_DATA['seq'])
+        # We took all elements so the view should match the original.
+        check_arr_seq(seq_view, SEQ_DATA['seq'])
+
+        # Get multiple items using ndarray of dtype integer.
+        for dtype in [np.int8, np.int16, np.int32, np.int64]:
+            seq_view = SEQ_DATA['seq'][np.array(indices, dtype=dtype)]
+            check_arr_seq_view(seq_view, SEQ_DATA['seq'])
+            # We took all elements so the view should match the original.
+            check_arr_seq(seq_view, SEQ_DATA['seq'])
+
+        # Get multiple items out of order (creates a view).
+        SEQ_DATA['rng'].shuffle(indices)
+        seq_view = SEQ_DATA['seq'][indices]
+        check_arr_seq_view(seq_view, SEQ_DATA['seq'])
+        check_arr_seq(seq_view, np.asarray(SEQ_DATA['data'])[indices])
```

Comment:
Judging by Travis, `np.asarray(SEQ_DATA['data'])` seems to rely on numpy versions more recent than 1.5.1.

    ValueError: setting an array element with a sequence.

Does it make more sense to use an alternative to `np.asarray()[indices]` to create the sequence you're looking for, or update the minimum numpy version ([1.5.1](https://github.com/numpy/numpy/releases/tag/v1.5.1) is over 5 years old)?